### ISSUE TITLE
Crash loop upgrading from 1.53.x to 1.54+

### ISSUE THREAD

**Issue by vmstan** [n8n-io/n8n#10420]
### Bug Description

I'm currently running n8n in a Kubernetes environment, with a dedicated external Postgres 16 database. I have been running 1.53.x attempting to move to 1.54.x or 1.55.x fails with a container crash loop with the following error:

```
2024-08-14 18:02:36.025Z,n8n-0,n8n,"User settings loaded from: /home/node/.n8n/config"
2024-08-14 18:02:39.589Z,n8n-0,n8n,"Last session crashed"
2024-08-14 18:02:49.598Z,n8n-0,n8n,"Initializing n8n process"
2024-08-14 18:02:50.925Z,n8n-0,n8n,"Error: There was an error initializing DB"
2024-08-14 18:02:50.925Z,n8n-0,n8n,"DatabaseError: no pg_hba.conf entry for host "10.118.0.7", user "n8n", database "n8n", no encryption"
```

Reverting the manifest back to 1.53.2 allows the pod to start. I reviewed the release notes for all versions between them, and didn't see anything about changes to Postgres configuration. I had a similar issue in #7413 that was resolved with SQL configuration options but not sure what to adjust this time.

Here is a redacted copy of the deployment manifest:

```
apiVersion: apps/v1
kind: StatefulSet
metadata:
  name: n8n
spec:
  selector:
    matchLabels:
      app: n8n
  serviceName: "n8n-service"
  replicas: 1
  template:
    metadata:
      labels:
        app: n8n
    spec:
      nodeSelector:
        node.kubernetes.io/instance-type: c-4
      containers:
        - name: n8n
          image: ghcr.io/n8n-io/n8n:1.53.2
          resources:
            requests:
              memory: "512Mi"
              cpu: "125m"
            limits:
              cpu: "500m"
              memory: "2G"
          ports:
            - containerPort: 5678
          volumeMounts:
            - mountPath: /home/node/.n8n
              name: data
            - mountPath: /root/do-certificate.crt
              subPath: do-certificate.crt
              name: do-certificate
              readOnly: true
          env:
            - name: GENERIC_TIMEZONE
              value: America/Chicago
            - name: TZ
              value: America/Chicago
            - name: N8N_ENCRYPTION_KEY
              value: xxxx
            - name: WEBHOOK_URL
              value: https://n8n.vmst.io/
            - name: N8N_PORT
              value: "5678"
            - name: N8N_PROTOCOL
              value: http
            - name: N8N_EMAIL_MODE
              value: smtp
            - name: N8N_SMTP_HOST
              value: xxxx
            - name: N8N_SMTP_SSL
              value: "true"
            - name: N8N_SMTP_PORT
              value: "465"
            - name: N8N_SMTP_USER
              value: xxxx
            - name: N8N_SMTP_PASS
              value: xxxx
            - name: N8N_SMTP_SENDER
              value: n8n@ses.vmst.io
            - name: DB_POSTGRESDB_DATABASE
              value: n8n
            - name: DB_POSTGRESDB_HOST
              value: xxxx
            - name: DB_POSTGRESDB_PORT
              value: "25060"
            - name: DB_POSTGRESDB_USER
              value: n8n
            - name: DB_POSTGRESDB_PASSWORD
              value: xxxx
            - name: DB_TYPE
              value: postgresdb
            - name: DB_POSTGRESDB_SSL_REJECT_UNAUTHORIZED
              value: "false"
            # - name: DB_POSTGRESDB_SSL_CA
            #   value: /root/do-certificate.crt
            # - name: DB_POSTGRESDB_SSL_ENABLED
            #   value: "true"
          livenessProbe:
            tcpSocket:
              port: 5678
          readinessProbe:
            httpGet:
              path: /healthz
              port: 5678
            timeoutSeconds: 1
            periodSeconds: 10
          startupProbe:
            httpGet:
              path: /healthz
              port: 5678
            failureThreshold: 30
            periodSeconds: 5
            initialDelaySeconds: 30
      restartPolicy: Always
      volumes:
        - name: data
          persistentVolumeClaim:
            claimName: data
        - name: do-certificate
          configMap:
            name: do-certificate
  volumeClaimTemplates:
    - metadata:
        name: data
      spec:
        accessModes: ["ReadWriteOnce"]
        resources:
          requests:
            storage: 2Gi
```

The commented out SSL settings were tried and resulted in different errors about a self-signed certificate being in the chain. No other database server configurations have been changed between the upgrade and downgrade attempts aside from the n8n version.

### To Reproduce

Attempt to upgrade from 1.53.x to 1.54.

### Expected behavior

n8n connects to the database

### Operating System

OS Containerized

### n8n Version

1.53 -> 1.54+

### Node.js Version

Node.js Version Containerized

### Database

PostgreSQL

### Execution mode

main (default)

**Comment by Joffcom** [comment]
Hey @vmstan,

We have created an internal ticket to look into this which we will be tracking as "GHC-162"

**Comment by Joffcom** [comment]
Hey @vmstan,

Just a quick thought on this one, Have you tried adding your self signed certificate to your container which would be safer than ignoring SSL issues?


https://docs.n8n.io/hosting/configuration/configuration-examples/custom-certificate-authority/

**Comment by vmstan** [comment]
Does the certificate need to be in a specific format? I've been trying to get this work and have the logs showing that it's trusting the files in that location but then still doesn't connect without `DB_POSTGRESDB_SSL_REJECT_UNAUTHORIZED` set to `false`.

**Comment by Joffcom** [comment]
Hey @vmstan,

Pem should do the job, I will get the doc updated to mention the write access as well.

**Comment by vmstan** [comment]
Doesn't seem to work, at least for me. The file is there in PEM format and accessible by the `node` user.

```
/opt/custom-certificates $ ls -lhsa
total 12K
   4.0K drwxr-xr-x    2 node     node        4.0K Aug 14 15:21 .
   4.0K drwxr-xr-x    1 root     root        4.0K Aug 14 15:21 ..
      0 lrwxrwxrwx    1 node     node          18 Aug 14 15:21 c1df2ce4.0 -> do-certificate.crt
   4.0K -rwxr-xr-x    1 node     node        1.5K Aug 14 15:21 do-certificate.crt
```

Still complains.

```
2024-08-14 20:20:52.509Z,n8n-0,n8n,"Trusting custom certificates from /opt/custom-certificates."
2024-08-14 20:20:54.189Z,n8n-0,n8n,"User settings loaded from: /home/node/.n8n/config"
2024-08-14 20:20:57.732Z,n8n-0,n8n,"Last session crashed"
2024-08-14 20:21:07.740Z,n8n-0,n8n,"Initializing n8n process"
2024-08-14 20:21:09.003Z,n8n-0,n8n,"Error: There was an error initializing DB"
2024-08-14 20:21:09.003Z,n8n-0,n8n,"Error: self-signed certificate in certificate chain"
```

**Comment by brandnewx** [comment]
Move your env variables to .env file for security reasons.

**Comment by DerEnderKeks** [comment]
I've got a very similar issue when doing the same version upgrade (1.53.2->1.54.3):

DB config:
```
DB_TYPE: postgresdb
DB_POSTGRESDB_DATABASE: n8n
DB_POSTGRESDB_HOST: [REDACTED]
DB_POSTGRESDB_PORT: '5432'
DB_POSTGRESDB_USER: {{ .db_username }}
DB_POSTGRESDB_PASSWORD: {{ .db_password }}
DB_POSTGRESDB_SSL_ENABLED: 'true'
```
(I use a Let's Encrypt cert for my DB, so no need to specify a custom CA)

Log output when starting version 1.54.3:

```
Initializing n8n process
Error: There was an error initializing DB
Error: unable to get local issuer certificate
```

**Comment by netroy** [comment]
This was likely due to a change in our config parser in 1.54.0, that was sometimes incorrectly parsing booleans and numbers.
We've fixed this (and added tests) in 1.54.4 and also in 1.55.1.

Can you folks please update and check if this fixes the issue for you? 🙏🏽

**Comment by vmstan** [comment]
1.55.1 worked for me. Thank you.
